### DISCUSSION TITLE
guide-R/analyses/statistique-univariee

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#20] [Forum]
# guide-R/analyses/statistique-univariee

https://larmarange.github.io/guide-R/analyses/statistique-univariee.html

**Comment by WilliamZ98** [comment]
Bonjour Monsieur,

Je souhaite vous poser une question concernant la présentation des modalités d'une variable catégorielle à travers un diagramme en barres. En l'absence de possibilité de les renommer lorsque les libellés sont longs, comment puis-je gérer leur affichage de manière à éviter qu'ils ne s'entremêlent ? Par exemple, est-il possible de les incliner de la même manière qu'on le fait dans Excel ?

Respectueusement !

**Reply by larmarange** [reply]
Bonjour, vous avez plusieurs possibilités, qui peuvent être combinées :

- passer en barres horizontales, ce qui laisse plus de place pour les intitulés
- utiliser `scales::label_wrap()` pour couper les intitulés trop longs sur plusieurs lignes
- vous pouvez également changer l'orientation des étiquettes avec `ggplot2::theme()` et `ggplot2::element_text()` dont l'argument `angle` permet de procéder à une orientation des étiquettes


Voir des exemples sur https://larmarange.github.io/analyse-R/exemples-graphiques-avances.html

et https://www.r-bloggers.com/2021/09/how-to-rotate-axis-labels-in-ggplot2/ pour la rotation des étiquettes

**Reply by WilliamZ98** [reply]
Excellent ! Merci monsieur.
